cmd/kubeadm/app/phases/kubelet/config.go

```diff
@@ -57,28 +62,34 @@ func CreateConfigMap(cfg *kubeadmapi.ClusterConfiguration, client clientset.Inte
 	}
 
 	configMapName := kubeadmconstants.GetKubeletConfigMapName(k8sVersion)
-	fmt.Printf("[kubelet] Creating a ConfigMap %q in namespace %s with the configuration for the kubelets in the cluster\n", configMapName, metav1.NamespaceSystem)
-
 	kubeletCfg, ok := cfg.ComponentConfigs[componentconfigs.KubeletGroup]
 	if !ok {
 		return errors.New("no kubelet component config found in the active component config set")
 	}
 
-	kubeletBytes, err := kubeletCfg.Marshal()
-	if err != nil {
-		return err
-	}
+	// If the component config is user supplied (loaded from a file or a config map) it's stored in a config map.
+	// If it's purely generated by kubeadm, we don't store it in the cluster (although we still setup RBAC correctly).
+	if kubeletCfg.UserSupplied() {
+		fmt.Printf("[kubelet] Creating a ConfigMap %q in namespace %s with the configuration for the kubelets in the cluster\n", configMapName, metav1.NamespaceSystem)
 
-	if err := apiclient.CreateOrUpdateConfigMap(client, &v1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      configMapName,
-			Namespace: metav1.NamespaceSystem,
-		},
-		Data: map[string]string{
-			kubeadmconstants.KubeletBaseConfigurationConfigMapKey: string(kubeletBytes),
-		},
-	}); err != nil {
-		return err
+		kubeletBytes, err := kubeletCfg.Marshal()
+		if err != nil {
+			return err
+		}
+
+		if err := apiclient.CreateOrUpdateConfigMap(client, &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      configMapName,
+				Namespace: metav1.NamespaceSystem,
+			},
+			Data: map[string]string{
+				kubeadmconstants.KubeletBaseConfigurationConfigMapKey: string(kubeletBytes),
+			},
+		}); err != nil {
+			return err
+		}
+	} else {
+		fmt.Printf("[kubelet] Skip creating ConfigMap %q in namespace %s: generated config\n", configMapName, metav1.NamespaceSystem)
```

Comment:
I fear the user might be worried to read "Skip creating" if we don't provide an explanation about why this is happening. WDYT?

Reply:
+1